enhancements/network/ovn-multi-external-gateway.md

```diff
@@ -0,0 +1,183 @@
+- Removal of the annotation based logic to configure multiple external gateways. This should be carrier out at least no sooner than OpenShift 4.15 to guarantee a stable deprecation path to existing deployments that leverage on the annotations to configure the external gateways.
+
+## Proposal
+
+Create a new CRD that contains a direct relationship to the annotations used in the pod/namespace to manage the external gateways configuration:
+
+Namespace annotations:
+* `k8s.ovn.org/routing-external-gws`
+* `k8s.ovn.org/external-gw-pod-ips`
+
+Pod annotations:
+* `k8s.ovn.org/routing-namespaces`
+* `k8s.ovn.org/routing-network`
+* `k8s.ovn.org/bfd-enabled`
+
+The implementation will also include a new controller that will watch cluster wide for instances of this CRD and apply the changes to the OVN network configuration as it is currently done by the annotation based logic.
+
+The controller will use the OpenShift release version as well as the existence of the annotations in the namespace and pods to apply the changes to the network configuration:
+
+| **OpenShift version** | **Annotations exist** | **CR exists** | **Action**                                        |
+|-----------------------|-----------------------|---------------|---------------------------------------------------|
+| 4.13                  | No                    | Yes           | Use CR information to configure external gateways |
+| 4.13                  | Yes                   | Yes           | Use annotations to configure external gateways    |
+| 4.13                  | Yes                   | No            | Use annotations to configure external gateway     |
+| 4.13                  | No                    | No            | Nothing to do                                     |
+| 4.14                  | No                    | Yes           | Use CR information to configure external gateways |
+| 4.14                  | Yes                   | Yes           | Use CR information to configure external gateways |
+| 4.14                  | Yes                   | No            | Use annotations to configure external gateways    |
+| 4.14                  | No                    | No            | Nothing to do                                     |
+
+The current annotation based logic will be enhanced as part of this change to support the logic described in the previous table.
+
+### Workflow Description
+The workflow follows the behavior of a namespaced CR.
+
+
+### API Extensions
+
+A new namespaced-scoped CRD is created to represent the multiple external gateways configuration. The annotations are mapped as following:
+* `k8s.ovn.org/routing-external-gws` is a namespace specific annotation and is represented in the CRD as `routingExternalGateways`. The values represent a slice of strings containing the IPs used for external traffic. 
+This annotation works in conjunction with `k8s.ovn.org/external-gw-pod-ips` and is used to capture the IPs that are stored in conntrack.
```

Comment:
external-gw-pod-ips tracks IPs of pods that are acting as external gateways. It does not store IPs in routing-external-gws.

Reply:
Thanks for the clarification. I will update

Reply:
As per @tssurya 's comment, this field is no longer relevant at this point, so I have not included it in the spec.

Reply:
@jordigilh : but this can be remove only if like I mentioned someone tests the OVN fix and removes these bits form OVNK if the OVN fix works. Until that is done, we must support this annotation.

Reply:
@trozet is there a plan to test the fix before this enhancement is implemented? If not, we'll have to update the annotation in both cases.

Reply:
@jordigilh : my two cents is you can do this as pre-work for the epic? I bet it will only simplify your epic so it will be an advantage for you, we could create a bz/card for this work and you can own it, outside of that it is currently not in the team's set of priorities AFAIK, I'll let @trozet chime in.